### DISCUSSION TITLE
Best way to send all command output to a fresh popup?

### DISCUSSION THREAD

**Discussion by jrnxf** [folke/noice.nvim#255] [Q&A]
I'd like for any command I type in the `cmdline` that produces actual output to be sent to a popup (or maybe a split - still undecided). Here's an example

1. Open neovim
2. Type `:hi`
3. popup displays all highlight groups
4. close popup
5. Type `:command`
6. popup displays all commands (nothing related to highlights)

I read through all of the documentation and the best I could come up with was, but that gives me messages I don't want (i.e. "yanked 7 lines") and the popup is never cleared, I always have to go to the bottom to see my latest output.

```
  routes = {
    {
      view = 'popup',
      filter = { event = 'msg_show' },
      opts = { enter = true },
    },
  },
```

Is this possible?

**Comment by folke** [comment]
There's a hidden preset you can use for this `cmdline_output_to_split=true`

See also my remarks on this here https://github.com/folke/noice.nvim/pull/256

**Reply by jrnxf** [reply]
I think I might be a tad confused. What does that neovim issue have to do with the ability to have a clean popup with no output from prior commands? I may have been unclear, essentially what I would like is for the ability to run command a, see the output of command a in a popup or split, close that popup or split, run command b, and see ONLY command b in the popup or split. Essentially what I would see if I were to run `:Noice history`
